src/pages/scans/ScansListView.tsx

```diff
@@ -27,37 +34,86 @@ import {
+import useScanApi from 'src/hooks/api/useScanApi';
+import useAlerts from 'src/hooks/useAlerts';
+import useQueryClientConfig from 'src/services/queryClientConfig';
 import { helpers } from '../../common';
 import { ContextIcon, ContextIconVariant } from '../../components/contextIcon/contextIcon';
 import { RefreshTimeButton } from '../../components/refreshTimeButton/RefreshTimeButton';
 import { ScanType } from '../../types';
 import { useScansQuery } from './useScansQuery';
 
-const SCANS_LIST_QUERY = 'scansList';
-
 const ScansListView: React.FunctionComponent = () => {
   const { t } = useTranslation();
-  const [alerts, setAlerts] = React.useState<Partial<AlertProps>[]>([]);
   const [refreshTime, setRefreshTime] = React.useState<Date | null>();
-  const [scanSelected, setScanSelected] = React.useState<ScanType>();
   const [scanSelectedForSources, setScanSelectedForSources] = React.useState<ScanType>();
-  const [pendingDeleteScan, setPendingDeleteScan] = React.useState<ScanType>();
-  const queryClient = useQueryClient();
+  const {
+    deleteScan,
+    onDeleteSelectedScans,
+    runScan,
+    pendingDeleteScan,
+    setPendingDeleteScan,
+    scanSelected,
+    setScanSelected
+  } = useScanApi();
+  const { queryClient } = useQueryClientConfig();
+  const { alerts, addAlert, removeAlert } = useAlerts();
 
   const onRefresh = () => {
-    queryClient.invalidateQueries({ queryKey: [SCANS_LIST_QUERY] });
+    /**
+     * Invalidates the query cache for the scan list, triggering a refresh.
+     */
+    queryClient.invalidateQueries({ queryKey: [API_SCANS_LIST_QUERY] });
   };
 
-  const addAlert = (title: string, variant: AlertProps['variant'], key: React.Key) => {
-    setAlerts(prevAlerts => [...prevAlerts, { title, variant, key }]);
+  const onDeleteScan = () => {
+    /**
+     * Deletes the pending scan and handles success, error, and cleanup operations.
+     */
+    deleteScan()
+      .then(() => {
+        addAlert(`Scan "${pendingDeleteScan?.id}" deleted successfully`, 'success', getUniqueId());
+        onRefresh();
+      })
+      .catch(err => {
+        console.log(err);
+        let errorDetail = '';
+        if (err?.response?.data) {
+          errorDetail += JSON.stringify(err.response.data);
+        }
+        addAlert(
+          `Error removing scan ${pendingDeleteScan?.id}. ${errorDetail}`,
+          'danger',
+          getUniqueId()
+        );
+      })
+      .finally(() => setPendingDeleteScan(undefined));
   };
 
-  const removeAlert = (key: React.Key) => {
-    setAlerts(prevAlerts => [...prevAlerts.filter(alert => alert.key !== key)]);
+  const onRunScan = (payload: ScanType) => {
+    /**
+     * Initiates a scan for the provided source, handles success, error, and cleanup operations.
+     *
+     * @param {ScanType} payload - The payload containing source information to start the scan.
+     */
+    console.log('run scan:', payload);
+    runScan(payload);
```

Comment:
the runScan rest call actually takes a list of IDs, as such we'll need to pass the id in here instead of the entire ScanType object. as such, you should probably change `payload` name to `scan` as it isn't the payload that is getting sent.. then do `scan.id` or whatever.